### DISCUSSION TITLE
Issue deploying github page due to Jekyll

### DISCUSSION THREAD

**Discussion by jacobhellenbrand** [community/community#108107] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I was addding information to my site 5 days ago and everythign was deploying correctly. I went to add some information to my github.io site today but I haven't been able to deploy all day. It is giving me this error: 

Error: undefined method `excerpt_separator' for #<Jekyll::Page @name="404.md"> 

but when checking my config.yml file and the 404 file everything seemes correct. Github is also telling me my version of bundler is 2.1.4 and I need 2.5.4, when i check vs code it says I have that and when I check my Mac Terminal it says I have 2.5.6. I'm also being told to install missing gems but when I run bundle they do not install, and if I install them individually the error does not change. 

I can attach the error message here: 

`Run actions/jekyll-build-pages@v1
/usr/bin/docker run --name ghcrioactionsjekyllbuildpagesv1010_745204 --label 63ce39 --workdir /github/workspace --rm -e "INPUT_SOURCE" -e "INPUT_DESTINATION" -e "INPUT_FUTURE" -e "INPUT_BUILD_REVISION" -e "INPUT_VERBOSE" -e "INPUT_TOKEN" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_ENVIRONMENT" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e "ACTIONS_ID_TOKEN_REQUEST_URL" -e "ACTIONS_ID_TOKEN_REQUEST_TOKEN" -e "ACTIONS_RESULTS_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/jacobhellenbrand.github.io/jacobhellenbrand.github.io":"/github/workspace" ghcr.io/actions/jekyll-build-pages:v1.0.10
Warning: the running version of Bundler (2.1.4) is older than the version that created the lockfile (2.5.4). We suggest you to upgrade to the version that created the lockfile by running `gem install bundler:2.5.4`.
Resolving dependencies...
The following gems are missing
 * uri (0.13.0)
 * net-http (0.4.1)
 * faraday-net_http (3.1.0)
 * faraday (2.9.0)
 * nokogiri (1.16.2)
 * hawkins (2.0.5)
Install missing gems with `bundle install`
Warning: The github-pages gem can't satisfy your Gemfile's dependencies. If you want to use a different Jekyll version or need additional dependencies, consider building Jekyll site with GitHub Actions: https://jekyllrb.com/docs/continuous-integration/github-actions/
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
/usr/local/bundle/gems/jekyll-3.9.4/lib/jekyll/excerpt.rb:135:in `extract_excerpt': undefined method `excerpt_separator' for #<Jekyll::Page @name="404.md"> (NoMethodError)
	from /usr/local/bundle/gems/jekyll-3.9.4/lib/jekyll/excerpt.rb:25:in `initialize'
	from /usr/local/bundle/gems/jekyll-relative-links-0.7.0/lib/jekyll-relative-links/generator.rb:173:in `new'
	from /usr/local/bundle/gems/jekyll-relative-links-0.7.0/lib/jekyll-relative-links/generator.rb:173:in `replace_relative_links_excerpt!'
	from /usr/local/bundle/gems/jekyll-relative-links-0.7.0/lib/jekyll-relative-links/generator.rb:64:in `replace_relative_links!'
	from /usr/local/bundle/gems/jekyll-relative-links-0.7.0/lib/jekyll-relative-links/generator.rb:44:in `block in generate'
	from /usr/local/bundle/gems/jekyll-relative-links-0.7.0/lib/jekyll-relative-links/generator.rb:39:in `each'
	from /usr/local/bundle/gems/jekyll-relative-links-0.7.0/lib/jekyll-relative-links/generator.rb:39:in `generate'
	from /usr/local/bundle/gems/jekyll-3.9.4/lib/jekyll/site.rb:175:in `block in generate'
	from /usr/local/bundle/gems/jekyll-3.9.4/lib/jekyll/site.rb:173:in `each'
	from /usr/local/bundle/gems/jekyll-3.9.4/lib/jekyll/site.rb:173:in `generate'
	from /usr/local/bundle/gems/jekyll-3.9.4/lib/jekyll/site.rb:70:in `process'
	from /usr/local/bundle/gems/jekyll-3.9.4/lib/jekyll/command.rb:28:in `process_site'
	from /usr/local/bundle/gems/jekyll-3.9.4/lib/jekyll/commands/build.rb:65:in `build'
	from /usr/local/bundle/gems/jekyll-3.9.4/lib/jekyll/commands/build.rb:36:in `process'
	from /usr/local/bundle/gems/github-pages-229/bin/github-pages:70:in `block (3 levels) in <top (required)>'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `block in execute'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `each'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `execute'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/program.rb:42:in `go'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary.rb:19:in `program'
	from /usr/local/bundle/gems/github-pages-229/bin/github-pages:6:in `<top (required)>'
	from /usr/local/bundle/bin/github-pages:23:in `load'
	from /usr/local/bundle/bin/github-pages:23:in `<main>'
Error:  Logging at level: debug Configuration file: /github/workspace/./_config.yml GitHub Pages: github-pages v229 GitHub Pages: jekyll v3.9.4 Theme: jekyll-theme-primer Theme source: /usr/local/bundle/gems/jekyll-theme-primer-0.6.0 Requiring: jekyll-github-metadata Requiring: jekyll-seo-tag Requiring: jekyll-paginate Requiring: jekyll-sitemap Requiring: jekyll-gist Requiring: jekyll-feed Requiring: jekyll-redirect-from Requiring: jekyll-coffeescript Requiring: jekyll-commonmark-ghpages Requiring: jekyll-github-metadata Requiring: jekyll-relative-links Requiring: jekyll-optional-front-matter Requiring: jekyll-readme-index Requiring: jekyll-default-layout Requiring: jekyll-titles-from-headings GitHub Metadata: Initializing... Source: /github/workspace/. Destination: /github/workspace/./_site Incremental build: disabled. Enable with --incremental Generating... EntryFilter: excluded /package.json EntryFilter: excluded /CHANGELOG.md EntryFilter: excluded /vendor EntryFilter: excluded /LICENSE EntryFilter: excluded /Gemfile EntryFilter: excluded /README.md EntryFilter: excluded /Gemfile.lock Reading: _posts/2013-08-14-blog-post-2.md Reading: _posts/2012-08-14-blog-post-1.md Reading: _posts/2015-08-14-blog-post-4.md Reading: _posts/2014-08-14-blog-post-3.md EntryFilter: excluded /assets/js/plugins EntryFilter: excluded /assets/js/vendor Reading: _teaching/2014-spring-teaching-1.md Reading: _teaching/2015-spring-teaching-2.md Reading: _publications/2015-10-01-paper-title-number-3.md Reading: _publications/2010-10-01-paper-title-number-2.md Reading: _publications/2009-10-01-paper-title-number-1.md Reading: _portfolio/portfolio-2.html Reading: _portfolio/portfolio-1.md Reading: _talks/2014-02-01-talk-2.md Reading: _talks/2012-03-01-talk-1.md Reading: _talks/2014-03-01-talk-3.md Reading: _talks/2013-03-01-tutorial-1.md Generating: JekyllRedirectFrom::Generator finished in 0.024001 seconds. Generating: JekyllOptionalFrontMatter::Generator finished in 0.00021372 seconds. Generating: JekyllReadmeIndex::Generator finished in 0.000394448 seconds. Generating: Jekyll::Paginate::Pagination finished in 4.969e-06 seconds. Generating: Jekyll::JekyllSitemap finished in 0.00029929 seconds. Jekyll Feed: Generating feed for posts Generating: JekyllFeed::Generator finished in 0.000687808 seconds. github-pages 229 | Error: undefined method `excerpt_separator' for #<Jekyll::Page @name="404.md"> `

This was also the first code edit I made when I had the deploy fail, these have been reverted since

<img width="1076" alt="Screenshot 2024-02-12 at 7 34 35 PM" src="https://github.com/community/community/assets/112660239/a1414e11-6bf7-4329-a10b-99bf51b1eacd">

Any help would be very appreciated

**Comment by stefanschwarzmueller** [comment]
I had the same problem recently and also still struggling

**Comment by subashbasnyat** [comment]
I had the same problem. 
So, locally I
1. Updated `ruby` to 3.0.0
2. Updated bundler.
3. When i tried to run locally, it showed there was issue `require': cannot load such file -- webrick bundle`.
So, I added the webrick gem
`bundle add webrick`
4. Tested locally. And pushed the changes to github.

Hope it helps!

**Reply by StupidBuluchacha** [reply]
bundle add webrick works! thx!

**Comment by chrisdel101** [comment]
adding webrick did not solve my issue.
`ruby 3.3.0`
